app/controllers/product_images_controller.rb

```diff
@@ -22,7 +22,7 @@ def product_params
   end
 
   def product_image_params
-    params.require(:product).permit(:images)
+    params.require(:product).permit({variants_attributes: [:images, :id]})
```

Comment:
Another small formatting change
```suggestion
    params.require(:product).permit({ variants_attributes: [:images, :id] })
```